rust/aptos-indexer-transaction-stream/src/transaction_stream.rs

```diff
@@ -582,42 +580,224 @@ pub async fn create_fetcher_loop(
+    }
+
+    pub async fn reconnect_to_grpc(&mut self) {
+        // Sleep for 100ms between reconnect tries
+        // TODO: Turn this into exponential backoff
+        tokio::time::sleep(Duration::from_millis(100)).await;
+
+        if self.reconnection_retries >= RECONNECTION_MAX_RETRIES {
+            error!(
+                processor_name = self.processor_name,
+                service_type = PROCESSOR_SERVICE_TYPE,
+                stream_address = self.indexer_grpc_data_service_address.to_string(),
+                "[Parser] Reconnected more than 100 times. Will not retry.",
+            );
+            panic!("[Parser] Reconnected more than 100 times. Will not retry.")
+        }
+        self.reconnection_retries += 1;
+        info!(
+            processor_name = self.processor_name,
+            service_type = PROCESSOR_SERVICE_TYPE,
+            stream_address = self.indexer_grpc_data_service_address.to_string(),
+            starting_version = self.next_version_to_fetch,
+            ending_version = self.request_ending_version,
+            reconnection_retries = self.reconnection_retries,
+            "[Parser] Reconnecting to GRPC stream"
+        );
+        let response = get_stream(
+            self.indexer_grpc_data_service_address.clone(),
+            self.indexer_grpc_http2_ping_interval,
+            self.indexer_grpc_http2_ping_timeout,
+            self.indexer_grpc_reconnection_timeout_secs,
+            self.next_version_to_fetch,
+            self.request_ending_version,
+            self.auth_token.clone(),
+            self.processor_name.to_string(),
+        )
+        .await;
+        let connection_id = match response.metadata().get(GRPC_CONNECTION_ID) {
+            Some(connection_id) => connection_id.to_str().unwrap().to_string(),
+            None => "".to_string(),
+        };
+        self.connection_id = Some(connection_id);
+        self.resp_stream = Some(response.into_inner());
+        info!(
+            processor_name = self.processor_name,
+            service_type = PROCESSOR_SERVICE_TYPE,
+            stream_address = self.indexer_grpc_data_service_address.to_string(),
+            connection_id = self.connection_id,
+            starting_version = self.next_version_to_fetch,
+            ending_version = self.request_ending_version,
+            reconnection_retries = self.reconnection_retries,
+            "[Parser] Successfully reconnected to GRPC stream"
+        );
+    }
+
+    pub async fn get_chain_id(self) -> u64 {
+        get_chain_id(
+            self.indexer_grpc_data_service_address,
+            self.indexer_grpc_http2_ping_interval,
+            self.indexer_grpc_http2_ping_timeout,
+            self.indexer_grpc_reconnection_timeout_secs,
+            self.auth_token,
+            self.processor_name,
+        )
+        .await
+    }
+}
+
+pub async fn create_fetcher_loop(
+    txn_sender: AsyncSender<TransactionsPBResponse>,
+    indexer_grpc_data_service_address: Url,
```

Comment:
Same sorta thing here, this dozen args could be just `config: TransactionStreamConfig`, amazing!